File: cards-service/src/main/java/a/gleb/cardservice/db/entity/CardEntity.java
package a.gleb.cardservice.db.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

import static jakarta.persistence.GenerationType.AUTO;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "user_card")
public class CardEntity {

    @Id
    @GeneratedValue(strategy = AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "card_title", nullable = false)
    private String cardTitle;

    @Column(name = "description")
    private String description;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "last_modified_date", nullable = false)
    private LocalDateTime lastModifiedDate;
}

File: card-service/src/main/kotlin/a/gleb/cardservice/db/entity/CardEntity.kt
package a.gleb.cardservice.db.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.AUTO
import jakarta.persistence.Id
import java.time.LocalDateTime
import java.util.*

@Entity(name = "user_card")
data class CardEntity(

    @Id
    @GeneratedValue(strategy = AUTO)
    var id: UUID?,

    @Column(name = "card_title", nullable = false)
    var title: String,

    @Column(name = "description", nullable = true)
    var description:  String?,

    @Column(name = "user_id", nullable = false)
    var userId: UUID,

    @Column(name = "last_modified_date", nullable = false)
    var lastUpdate: LocalDateTime
)
